crates/biome_js_analyze/src/lint/correctness/use_block_scoped_var.rs

```diff
@@ -0,0 +1,229 @@
+use crate::services::semantic::SemanticServices;
+use biome_analyze::{Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule};
+use biome_console::markup;
+use biome_diagnostics::Severity;
+use biome_js_syntax::binding_ext::{AnyJsBindingDeclaration, AnyJsIdentifierBinding};
+use biome_rowan::{AstNode, SyntaxNode, TextRange};
+
+declare_lint_rule! {
+    /// Enforce the use of variables within the scope they are defined.
+    ///
+    /// This rule generates warnings when variables are used outside of the block
```

Comment:
```suggestion
    /// This rule detects when variables are used outside of the block
```
 Better use a wording that is different from the concept of severity, otherwise it could confuse users